src/inmanta/deploy/tasks.py

```diff
@@ -249,29 +249,24 @@ async def execute(self, task_manager: "scheduler.TaskManager", agent: str, reaso
             my_executor: executor.Executor = await self.get_executor(
                 task_manager, agent, executor_resource_details.id.entity_type, self.version
             )
-
-            dryrun_result: executor.DryrunResult = await my_executor.dry_run(executor_resource_details, self.dry_run_id)
-            await task_manager.dryrun_update(
-                env=task_manager.environment,
-                dryrun_result=dryrun_result,
-            )
-
         except Exception:
-            # FIXME: seems weird to conclude undeployable state from generic Exception on either of two method calls
             logger_for_agent(agent).error(
                 "Skipping dryrun for resource %s because it is in undeployable state",
                 executor_resource_details.rvid,
                 exc_info=True,
             )
-            result = executor.DryrunResult(
+            dryrun_result = executor.DryrunResult(
                 rvid=executor_resource_details.rvid,
                 dryrun_id=self.dry_run_id,
                 changes={"handler": AttributeStateChange(current="FAILED", desired="Resource is in an undeployable state")},
                 started=started,
                 finished=datetime.datetime.now().astimezone(),
                 messages=[],
             )
-            await task_manager.dryrun_update(env=task_manager.environment, dryrun_result=result)
+        else:
+            dryrun_result = await my_executor.dry_run(executor_resource_details, self.dry_run_id)
```

Comment:
I think this code needs a try except as well, to make sure all exceptions are handled

Reply:
Added try except here and changed the error description for the one above. What do you think? I did not want to have the same error log in both cases